packages/expo-updates/android/src/main/java/expo/modules/updates/launcher/DatabaseLauncher.kt

```diff
@@ -146,22 +141,25 @@ class DatabaseLauncher(
   }
 
   private fun embeddedAssetFileMap(context: Context): MutableMap<AssetEntity, String> {
-    val embeddedManifest = EmbeddedManifest.get(context, this.configuration)
-    val embeddedAssets: List<AssetEntity> = embeddedManifest?.assetEntityList ?: listOf()
+    val embeddedManifest = EmbeddedManifest.get(context, configuration)
+    val embeddedAssets = embeddedManifest?.assetEntityList ?: listOf()
+    logger?.info("embeddedAssetFileMap: embeddedAssets count = ${embeddedAssets.count()}")
     return mutableMapOf<AssetEntity, String>().apply {
       for (asset in embeddedAssets) {
         if (asset.isLaunchAsset) {
           continue
         }
-        val filename = asset.relativePath
-        if (filename != null) {
-          val embeddedAssetFilename = asset.embeddedAssetFilename
-          val file = if (embeddedAssetFilename != null) {
-            File(embeddedAssetFilename)
-          } else {
-            File(updatesDirectory, asset.relativePath!!)
-          }
-          this[asset] = Uri.fromFile(file).toString()
+        val filename = UpdatesUtils.createFilenameForAsset(asset)
+        asset.relativePath = filename
+        val assetFile = File(updatesDirectory, filename)
+        if (!assetFile.exists()) {
+          loaderFiles.copyAssetAndGetHash(asset, assetFile, context)
+        }
+        if (assetFile.exists()) {
+          this[asset] = Uri.fromFile(assetFile).toString()
+          logger?.info("embeddedAssetFileMap: ${asset.key},${asset.type} => ${this[asset]}")
+        } else {
+          logger?.error("embeddedAssetFileMap: no file for ${asset.key},${asset.type}", UpdatesErrorCode.AssetsFailedToLoad)
```

Comment:
in what case would this occur? seems like `copyAssetAndGetHash` shouldn't fail?

Reply:
I don't know how it would fail for the embedded update, but did not want it to fail silently.